src/Language/PureScript/PSString.hs

```diff
@@ -163,8 +201,8 @@ prettyPrintStringJS :: PSString -> Text
 prettyPrintStringJS s = "\"" <> foldMap encodeChar (toUTF16CodeUnits s) <> "\""
   where
   encodeChar :: Word16 -> Text
-  encodeChar c | c > 0xFF = "\\u" <> hex 4 c
-  encodeChar c | c > 0x7E || c < 0x20 = "\\x" <> hex 2 c
+  encodeChar c | c > 0xFF = "\\u" <> showHex' 4 c
```

Comment:
Should this have `{}` delimiters?

Reply:
Oh, good spot, yes! This won’t handle anything above U+FFFF correctly as it is now. I think I should add a test for this.

Reply:
Oh wait no - this is okay as-is, because `c` is a UTF-16 code unit (Word16) here, i.e. it can't be any larger than 0xFFFF. So the `4` in `showHex' 4` is ensuring we have at least 4 hex digits, and the fact that `c` is a `Word16` ensures that we have at most 4 hex digits. This function will render astral plane characters as surrogate pairs, e.g. U+1F431 -> `"\uD38D\uDC31"`

Reply:
`{}` delimiters probably wouldn't hurt though.